extension/src/paymentHandler/payment-handler.js

```diff
@@ -20,22 +22,58 @@ const PAYMENT_METHOD_TYPE_KLARNA_METHODS = [
   'klarna_account',
 ]
 
-async function handlePayment(paymentObject) {
+function _validatePaymentRequest(paymentObject, authToken) {
+  if (!auth.isBasicAuthEnabled()) {
+    const paymentValidator = ValidatorBuilder.withPayment(paymentObject)
+      .validateMetadataFields()
+      .validateRequestFields()
+      .validateReference()
+      .validateAmountPlanned()
+    if (paymentValidator.hasErrors())
+      return {
+        success: false,
+        data: paymentValidator.buildCtpErrorResponse(),
+      }
+  } else {
+    const paymentValidator = ValidatorBuilder.withPayment(paymentObject)
+    paymentValidator.validateMetadataFields()
+    if (paymentValidator.hasErrors())
```

Comment:
Didn't we agree that we should first validate the authorization?

Reply:
we need project key to retrieve credential stored in config. We agree that not using `username` + `project key `concatenation approach. Therefore, at least we have to validate the metadata fields to guarantee project key is found in payment object.

Reply:
I thought we did this global `basicAuth` option exactly for this reason, so we don't have to retrieve the key but first check if `basicAuth` is on. If so, the username and password must be provided (regardless if they're correct or not). If they're not provided, you should throw Unauthorized error.

Reply:
I thought we have to validate authorization header correctness and existence together.

https://github.com/commercetools/commercetools-adyen-integration/pull/584/commits/6a5a514c990a2d973b873458bf1c8b4067c953ce

Reply:
Last commit does not seem to fix the issue completely. If basic auth is active and auth token is supplied within the header but is wrong we would first validate validateMetadataFields and return some other validation instead of 403?